File: ios/GreatPeople/Models/Card.swift
import Foundation

struct Card: Identifiable, Codable, Equatable, Hashable {
    let id: String
    let figureName: String
    let era: String
    let gender: String
    let identities: [String]
    let lore: String
    let portraitUrl: String
    let years: String
    let trait: String
    let achievement: String
    let politics: Int
    let strength: Int
    let culture: Int
    let wealth: Int
    let intelligence: Int
    let technique: Int
    let belief: Int
    let reputation: Int

    var portraitURL: URL? {
        portraitUrl.hasPrefix("asset:") ? nil : URL(string: portraitUrl)
    }
    var localAssetName: String? {
        portraitUrl.hasPrefix("asset:") ? String(portraitUrl.dropFirst(6)) : nil
    }
    var totalStats: Int { politics + strength + culture + wealth + intelligence + technique + belief + reputation }
}

File: android/app/src/main/kotlin/com/greatpeople/card/data/model/Card.kt
package com.greatpeople.card.data.model

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.greatpeople.card.data.local.StringListConverter
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(tableName = "cards")
@TypeConverters(StringListConverter::class)
data class Card(
    @PrimaryKey val id: String,
    val figureName: String,
    val era: String,
    val gender: String = "",
    val identities: List<String> = emptyList(),
    val lore: String,
    val portraitUrl: String,
    val years: String = "",
    val trait: String = "",
    val achievement: String = "",
    val politics: Int = 0,
    val strength: Int = 0,
    val culture: Int = 0,
    val wealth: Int = 0,
    val intelligence: Int = 0,
    val technique: Int = 0,
    val belief: Int = 0,
    val reputation: Int = 0,
) : Parcelable
